packages/cactus-cmd-api-server/src/test/typescript/unit/get-open-api-spec-v1-oauth2-scopes.test.ts

```diff
@@ -0,0 +1,165 @@
+
+    apiServer = new ApiServer({
+      config: config.getProperties(),
+      pluginRegistry,
+    });
+
+    apiServer.initPluginRegistry({ pluginRegistry });
+    const startResponsePromise = apiServer.start();
+    await expect(startResponsePromise).toResolve();
+    const startResponse = await startResponsePromise;
+    expect(startResponse).toBeTruthy();
+
+    const { addressInfoApi } = await startResponsePromise;
+    const protocol = apiSrvOpts.apiTlsEnabled ? "https" : "http";
+    const { address, port } = addressInfoApi;
+    const apiHost = `${protocol}://${address}:${port}`;
+
+    const jwtPayload = { name: "Peter", location: "Albertirsa" };
+    const validJwt = await new SignJWT(jwtPayload)
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer(expressJwtOptions.issuer)
+      .setAudience(expressJwtOptions.audience)
+      .sign(jwtKeyPair.privateKey);
+    expect(validJwt).toBeTruthy();
+
+    const validBearerToken = `Bearer ${validJwt}`;
+    expect(validBearerToken).toBeTruthy();
+
+    apiClient = new ApiServerApiClient(
+      new ApiServerApiClientConfiguration({
+        basePath: apiHost,
+        baseOptions: { headers: { Authorization: validBearerToken } },
+        logLevel,
+      }),
+    );
+  });
+
+  it("HTTP - returns the OpenAPI spec .json document of the API server itself", async () => {
+    const res1Promise = apiClient.getOpenApiSpecV1();
+    await expect(res1Promise).resolves.toHaveProperty("data.openapi");
+    const res1 = await res1Promise;
+    expect(res1.status).toEqual(200);
+    expect(res1.data).toBeTruthy();
+
+    console.log("Response data type:", typeof res1.data);
+    console.log("Response data:", res1.data);
+
+    let openApiSpec;
+    try {
+      openApiSpec =
+        typeof res1.data === "string" ? JSON.parse(res1.data) : res1.data;
+    } catch (error) {
+      throw new Error(`Failed to parse OpenAPI spec: ${error.message}`);
+    }
+
+    expect(openApiSpec).toHaveProperty("components");
+    expect(openApiSpec.components).toHaveProperty("securitySchemes");
+
+    const securitySchemes = openApiSpec.components.securitySchemes;
+    expect(securitySchemes).toBeObject();
+
+    const expectedScopes: IExpectedScopes = {
+      "read:health": "Read health information",
+      "read:metrics": "Read metrics information",
+      "read:spec": "Read OpenAPI specification",
+    };
+
+    const securitySchemeNames = Object.keys(securitySchemes);
+
+    securitySchemeNames.forEach((schemeName) => {
+      const scheme = securitySchemes[schemeName];
+      expect(scheme).toHaveProperty("flows");
+      const flows = scheme.flows;
+      expect(flows).toHaveProperty("authorizationCode");
+      const scopes = flows.authorizationCode.scopes as IExpectedScopes;
+
+      Object.keys(expectedScopes).forEach((scope) => {
+        expect(scopes).toHaveProperty(scope);
+        expect(scopes[scope]).toEqual(expectedScopes[scope]);
+      });
+    });
+  });
```

Comment:
@aldousalvarez This verifies that the OpenAPI .json file has the security scheme declared.  It does not verify that those are being respected by the endpoint implementations themselves. Please update the endpoints touched by this diff to make use of the declared scopes and then write **2 more test cases for each endpoint** in addition this one I'm commenting on. One of them should be the positive test case where with a valid JWT the API client can execute the request and with an invalid JWT the request execution fails with an HTTP 401 unauthorized error.
Note: by test cases I just mean another `it("does something...", async () => {...})` block not a separate file.

Reply:
@aldousalvarez Thank you